src/utils/MainApi.js

```diff
@@ -0,0 +1,75 @@
+const BASE_URL = 'https://api.alexandra.gritsenko.students.nomoredomainssbs.ru';
+// const BASE_URL = 'http://localhost:3000';
+const handleResponse = (res) =>
+  res.ok ? res.json() : Promise.reject(`Error ${res.status}`);
+
+export const register = (email, password, name) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password, name }),
+  }).then(handleResponse);
```

Comment:
**Might be better**

You can make a speacial function `request` with requests handling not to duplicate it in every request

```js

  function request(url, options) {
    return fetch(url, options).then(handleResponse)
  }

```

So, your code will be like that

```js

const handleResponse = (res) =>
  res.ok ? res.json() : Promise.reject(`Error ${res.status}`);

  function request(url, options) {
    return fetch(url, options).then(handleResponse)
  }

export const register = (email, password, name) => {
  return request(`${BASE_URL}/signup`, {  // <== here you use `request` instead of `fetch`
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email, password, name }),
  })
};

export const login = (email, password) => {
  return request(`${BASE_URL}/signin`, {  // <== here you use `request` instead of `fetch`
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email, password }),
  })
};


```


and so on